src/main/java/com/example/demo/service/DoctorService.java

```diff
@@ -0,0 +1,99 @@
+package com.example.demo.service;
+
+import com.example.demo.exception.DoctorNotFoundException;
+import com.example.demo.exception.InvalidEmailException;
+import com.example.demo.exception.UserAlreadyExistsException;
+import com.example.demo.model.dto.CreateDoctorCommandDTO;
+import com.example.demo.model.dto.DoctorDTO;
+import com.example.demo.model.dto.EditDoctorCommandDTO;
+import com.example.demo.model.entity.Doctor;
+import com.example.demo.model.entity.User;
+import com.example.demo.model.mapper.DoctorMapper;
+import com.example.demo.repository.DoctorRepository;
+import com.example.demo.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.List;
+
+
+@Service
+@RequiredArgsConstructor
+public class DoctorService {
+    private final DoctorRepository doctorRepository;
+    private final DoctorMapper doctorMapper;
+    private final UserRepository userRepository;
+
+    public List<DoctorDTO> getAllDoctors() {
+        return doctorRepository.findAll().stream()
+                .map(doctorMapper::doctorToDoctorDTO)
+                .collect(Collectors.toList());
+    }
+
+    public DoctorDTO getDoctor(Long id) {
+        Doctor doctor = doctorRepository.findById(id)
+                .orElseThrow(() -> new DoctorNotFoundException("Doctor with the provided ID does not exist"));
+        return doctorMapper.doctorToDoctorDTO(doctor);
+    }
+
+    @Transactional
+    public DoctorDTO addNewDoctor(CreateDoctorCommandDTO command) {
+        String email = command.getEmail();
+        if (email == null || email.isEmpty()) {
+            throw new InvalidEmailException("Invalid email address");
+        }
+        Optional<User> existingDoctor = userRepository.findByEmail(email);
+        if (existingDoctor.isPresent()) {
+            throw new UserAlreadyExistsException("User with the provided email already exists");
+        }
+        Doctor doctor = Doctor.create(command);
+        doctorRepository.save(doctor);
```

Comment:
to zwraca Encje. Aby zrobic to najbardziej legitnie to powinnismy pomyslec czy nie lepiej byloby przyjac wynik:
 Doctor entity = doctorRepository.save(doctor);|
i do mappera przekazac tego zwroconego przez repozytorium dzieki temu nasz wynik bedzie najblizszy faktycznemu wynikowi tego co sie dodalo.